common/inventory-utils/src/semvrs.rs

```diff
@@ -1,7 +1,9 @@
-use crate::inv::VersionRequirement;
+use crate::inv::{Version, VersionRequirement};
 
 impl VersionRequirement<semver::Version> for semver::VersionReq {
     fn satisfies(&self, version: &semver::Version) -> bool {
         self.matches(version)
     }
 }
+
+impl Version for semver::Version {}
```

Comment:
I hear you, and actually had initially implemented the solution without a `Version` "marker trait" (early on when pairing with @Malax). The `Version` trait simply serves to signal that `Artifact#version` requires the `DeserializeOwned` and `Serialized` traits for serialization, but there's no reason we can't just bind those directly without a supertrait.

There's no plan for anything that needs to be there, and I've pushed a commit to show how this could be implemented to avoid the `Version` trait altogether here https://github.com/heroku/buildpacks-go/pull/254/commits/0e8fe280008495da40ef4052079c22ffc3ce7148. Do you like that solution better?